File: user-svc/UserService/Libs/Http/HttpServe.cs
namespace UserService.Libs.Http
{
    public class HttpServe<T>
    {
        public int Status { get; set; }
        public string Message { get; set; }
        public T? Data { get; set; }

        public HttpServe(int status, string message, T data)
        {
            Status = status;
            Message = message;
            Data = data;
        }
    }
}

File: order-svc/pkgs/http/httpserve/httpserve.go
package httpserve

// BaseResponse is a generic response structure
type BaseResponse[T any] struct {
	Code    int    `json:"code"`
	Message string `json:"message"`
	Data    T      `json:"data"`
}

// NewBaseResponse creates a new BaseResponse instance
func NewBaseResponse[T any](code int, message string, data T) BaseResponse[T] {
	return BaseResponse[T]{
		Code:    code,
		Message: message,
		Data:    data,
	}
}
